src/GafferBindings/ValuePlugBinding.cpp

```diff
@@ -220,17 +222,29 @@ void ValuePlugSerialiser::moduleDependencies( const Gaffer::GraphComponent *grap
 {
 	PlugSerialiser::moduleDependencies( graphComponent, modules, serialisation );
 
-	const ValuePlug *valuePlug = static_cast<const ValuePlug *> ( graphComponent );
-	object pythonPlug( ValuePlugPtr( const_cast<ValuePlug *>( valuePlug ) ) );
-	if( PyObject_HasAttrString( pythonPlug.ptr(), "defaultValue" ) )
+	// Get the module dependencies for the value held by the plug, in case we
+	// want to serialise a `getValue()` call or constructor with a default value.
+	// This is relatively expensive so we cache the results based on plug type.
+	// The cache is not threadsafe but we are protected by the GIL.
+	/// \todo This isn't quite right - ObjectPlug and ObjectVectorPlug might have
+	/// values of _any_ type on a per-plug basis, and the current value may not
+	/// have the same module dependencies as the default value.
+	static std::unordered_map<IECore::TypeId, std::string> g_cache;
+	auto inserted = g_cache.insert( { graphComponent->typeId(), std::string() } );
+	if( inserted.second )
 	{
-		object pythonDefaultValue = pythonPlug.attr( "defaultValue" )();
-		std::string module = Serialisation::modulePath( pythonDefaultValue );
-		if( module.size() )
+		const ValuePlug *valuePlug = static_cast<const ValuePlug *> ( graphComponent );
+		object pythonPlug( ValuePlugPtr( const_cast<ValuePlug *>( valuePlug ) ) );
+		if( PyObject_HasAttrString( pythonPlug.ptr(), "defaultValue" ) )
 		{
-			modules.insert( module );
+			object pythonDefaultValue = pythonPlug.attr( "defaultValue" )();
+			inserted.first->second = Serialisation::modulePath( pythonDefaultValue );
 		}
 	}
+	if( !inserted.first->second.empty() )
+	{
+		modules.insert( inserted.first->second );
+	}
```

Comment:
Much better idea, courtesy of @themissingcow. Let `constructor()`, `postHierarchy()` etc just split out module dependencies via a new `Serialiser::addModuleDependency()` method, so we don't have to figure it out separately. I'll give this a go and update the PR if it works out.